contracts/libraries/HyperCoreLib.sol

```diff
@@ -78,34 +74,28 @@ library HyperCoreLib {
      * @param to The address to receive tokens on HyperCore
      * @param amountEVM The amount to transfer on HyperEVM
      * @param decimalDiff The decimal difference of evmDecimals - coreDecimals
-     * @return amountCore The amount credited on Core in Core units (post conversion)
+     * @return amountEVMSent The amount sent on HyperEVM
+     * @return amountCoreToReceive The amount credited on Core in Core units (post conversion)
      */
     function transferERC20EVMToCore(
         address erc20EVMAddress,
         uint64 erc20CoreIndex,
         address to,
         uint256 amountEVM,
         int8 decimalDiff
-    ) internal returns (uint64 amountCore) {
+    ) internal returns (uint256 amountEVMSent, uint64 amountCoreToReceive) {
         // if the transfer amount exceeds the bridge balance, this wil revert
-        HyperAssetAmount memory amounts = quoteHyperCoreAmount(
-            erc20CoreIndex,
-            decimalDiff,
-            toAssetBridgeAddress(erc20CoreIndex),
-            amountEVM
-        );
-
-        if (amounts.evm != 0) {
+        (uint256 _amountEVMToSend, uint64 _amountCoreToReceive) = maximumEVMSendAmountToAmounts(amountEVM, decimalDiff);
```

Comment:
do we even want to modify the EVM send amount here? or rely on the higher level contract to do this?